cache/lru_cache.cc

```diff
@@ -662,6 +715,36 @@ double LRUCache::GetHighPriPoolRatio() {
   return result;
 }
 
+void LRUCache::WaitAll(std::vector<Handle*>& handles) {
+  if (secondary_cache_) {
+    std::vector<SecondaryCacheResultHandle*> sec_handles;
+    sec_handles.reserve(handles.size());
+    for (Handle* handle : handles) {
+      if (!handle) {
+        continue;
+      }
+      LRUHandle* lru_handle = reinterpret_cast<LRUHandle*>(handle);
+      if (!lru_handle->IsPending()) {
+        continue;
+      }
+      sec_handles.emplace_back(lru_handle->sec_handle);
+    }
+    secondary_cache_->WaitAll(sec_handles);
+    for (Handle* handle : handles) {
+      if (!handle) {
+        continue;
+      }
+      LRUHandle* lru_handle = reinterpret_cast<LRUHandle*>(handle);
+      if (!lru_handle->IsPending()) {
```

Comment:
Is it possible that, between L727 and L738, some of the flags of the handle in the handles are changed? So here we only promote the one that is still pending.

Reply:
Its not possible, because at this point, none of the handles have been inserted in the ```LRUHandleTable```, so no other lookups can get to it. Only ```WaitAll``` will change the handle from pending to not pending. I can add some comments to make it explicit that the handle is not thread safe.